### DISCUSSION TITLE
Roundtrip export and import of Dataset object to maintain same structure?

### DISCUSSION THREAD

**Discussion by Ixiodor** [corteva/rioxarray#249] [Q&A]
xarray let you associate some DataArrays to names in Datasets, saving a Dataset with "to_raster" those names become bands "long_name" parameters on "open_rasterio". There is a way to acess those bands trought definied names(after open_rasterio)?
Or using a Dataset(to_dataset) to map those names again as pre-saved item?

This is repr of readed object(open_rasterio):

> <xarray.DataArray (band: 4, y: 8944, x: 9916)>
> [354754816 values with dtype=float64]
> Coordinates:
>   * band         (band) int64 1 2 3 4
>   * y            (y) float64 4.923e+06 4.923e+06 ... 4.834e+06 4.834e+06
>   * x            (x) float64 6.271e+05 6.271e+05 ... 7.263e+05 7.263e+05
>     spatial_ref  int64 0
> Attributes:
>     _FillValue:    0.0
>     scale_factor:  1.0
>     add_offset:    0.0
>     long_name:     ('blue', 'green', 'red', 'nir')
>     grid_mapping:  spatial_ref

**Comment by snowman2** [comment]
It is stored in the attribute `long_name`. As shown in your `repr`:
```
long_name: ('blue', 'green', 'red', 'nir')
```

`xarray` is designed around the `netCDF` file structure. For best results in obtaining the same representation when opening, I would recommend using `to_netcdf`. Then, when opening the file with `open_dataset` or `open_rasterio` it will look the same as when you exported it.

Related to #171

**Reply by Ixiodor** [reply]
Yea but i mean:
no ways to do something like:
`red_band = raster_img.red`
or
`red_band = raster_img['red']`
behaviour like Dataset do?
Should i map manually index from long_name to band index?

**Reply by snowman2** [reply]
I updated the answer. Please see the part about `netCDF`

**Reply by snowman2** [reply]
> Should i map manually index from long_name to band index?

If you want a GeoTiff or similar file, then you will need to manually convert to a Dataset using the `long_name` and band index.

**Comment by snowman2** [comment]
See #296 #600
